docs/core/logging.md

```diff
@@ -377,320 +608,532 @@ for known event sources, where either a request ID or X-Ray Trace ID are present
 
-<!-- markdownlint-disable MD013 -->
-With [AWS Lambda Advanced Logging Controls (ALC)](https://docs.aws.amazon.com/lambda/latest/dg/monitoring-cloudwatchlogs.html#monitoring-cloudwatchlogs-advanced){target="_blank"}, you can enforce a minimum log level that Lambda will accept from your application code.
+???+ note "Note: Any object key named with `-` must be escaped"
+    For example, **`request.headers."x-amzn-trace-id"`**.
 
-When enabled, you should keep `Logger` and ALC log level in sync to avoid data loss.
+| Name                          | Expression                          | Description                     |
+|-------------------------------|-------------------------------------|---------------------------------|
+| **API_GATEWAY_REST**          | `"requestContext.requestId"`        | API Gateway REST API request ID |
+| **API_GATEWAY_HTTP**          | `"requestContext.requestId"`        | API Gateway HTTP API request ID |
+| **APPSYNC_RESOLVER**          | `request.headers."x-amzn-trace-id"` | AppSync X-Ray Trace ID          |
+| **APPLICATION_LOAD_BALANCER** | `headers."x-amzn-trace-id"`         | ALB X-Ray Trace ID              |
+| **EVENT_BRIDGE**              | `"id"`                              | EventBridge Event ID            |
 
-Here's a sequence diagram to demonstrate how ALC will drop both `INFO` and `DEBUG` logs emitted from `Logger`, when ALC log level is stricter than `Logger`.
-<!-- markdownlint-enable MD013 -->
 
-```mermaid
-sequenceDiagram
-    participant Lambda service
-    participant Lambda function
-    participant Application Logger
+## Customising fields in logs
 
-    Note over Lambda service: AWS_LAMBDA_LOG_LEVEL="WARN"
-    Note over Application Logger: POWERTOOLS_LOG_LEVEL="DEBUG"
+Powertools for AWS Lambda comes with default json structure ([standard fields](#standard-structured-keys) & [lambda context fields](#logging-lambda-context-information)).
 
-    Lambda service->>Lambda function: Invoke (event)
-    Lambda function->>Lambda function: Calls handler
-    Lambda function->>Application Logger: logger.error("Something happened")
-    Lambda function-->>Application Logger: logger.debug("Something happened")
-    Lambda function-->>Application Logger: logger.info("Something happened")
-    Lambda service--xLambda service: DROP INFO and DEBUG logs
-    Lambda service->>CloudWatch Logs: Ingest error logs
+You can go further and customize which fields you want to keep in your logs or not. The configuration varies according to the underlying logging library.
+
+### Log4j2 configuration
+Log4j2 configuration is done in _log4j2.xml_ and  leverages `JsonTemplateLayout`:
+
+```xml
+    <Console name="console" target="SYSTEM_OUT">
+        <JsonTemplateLayout eventTemplateUri="classpath:LambdaJsonLayout.json" />
+    </Console>
 ```
 
-### Priority of log level settings in Powertools for AWS Lambda
+The `JsonTemplateLayout` is configured with the provided template:
```

Comment:
This isn't a call-for-action to the user, right, it's happening automatically?

Reply:
```suggestion
The `JsonTemplateLayout` is automatically configured with the provided template:
```